x-pack/plugins/observability/server/lib/annotations/create_annotations_client.ts

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { APICaller, Logger } from 'kibana/server';
+import * as t from 'io-ts';
+import { SearchResponse, Client } from 'elasticsearch';
+import {
+  createAnnotationRt,
+  searchAnnotationsRt,
+  deleteAnnotationRt,
+  Annotation,
+  getAnnotationByIdRt,
+} from '../../../common/annotations';
+import { PromiseReturnType } from '../../../../apm/typings/common';
+import { createOrUpdateIndex } from '../../utils/create_or_update_index';
+
+type SearchParams = t.TypeOf<typeof searchAnnotationsRt>;
+type CreateParams = t.TypeOf<typeof createAnnotationRt>;
+type DeleteParams = t.TypeOf<typeof deleteAnnotationRt>;
+type GetByIdParams = t.TypeOf<typeof getAnnotationByIdRt>;
+
+interface IndexDocumentResponse {
+  _shards: {
+    total: number;
+    failed: number;
+    successful: number;
+  };
+  _index: string;
+  _type: string;
+  _id: string;
+  _version: number;
+  _seq_no: number;
+  _primary_term: number;
+  result: string;
+}
+
+export function createAnnotationsClient(params: {
+  index: string;
+  apiCaller: APICaller;
+  logger: Logger;
+}) {
+  const { index, apiCaller, logger } = params;
+
+  const initIndex = () =>
+    createOrUpdateIndex({
+      index,
+      mappings: {
+        dynamic: 'strict',
+        properties: {
+          annotation: {
+            properties: {
+              type: {
+                type: 'keyword',
+              },
+            },
+          },
+          message: {
+            type: 'text',
+          },
+          tags: {
+            type: 'keyword',
+          },
+          '@timestamp': {
+            type: 'date',
+          },
+          event: {
+            properties: {
+              created: {
+                type: 'date',
+              },
+            },
+          },
+          service: {
+            properties: {
+              name: {
+                type: 'keyword',
+              },
+              environment: {
+                type: 'keyword',
+              },
+              version: {
+                type: 'keyword',
+              },
+            },
+          },
+        },
+      },
+      apiCaller,
+      logger,
+    });
+
+  return {
+    search: async (searchParams: SearchParams): Promise<SearchResponse<Annotation>> => {
```

Comment:
Right now this is just a thin wrapper around `_search`. I've added it because we are planning on querying multiple sources in the near future (for instance, the event log). I was talking to @sqren about this and we wonder if we should even have it, or expect consumers of annotations to query the index directly. Having validation when indexing documents is definitely valuable but do we need a (limiting) abstraction around searching as well?

Reply:
I think it's better to remove it until there is a need.

Reply:
removed, APM now talks to ES directly to get the stored annotations